packages/ckeditor5-dev-utils/tests/translations/retryasyncfunction.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const mockery = require( 'mockery' );
+const chai = require( 'chai' );
+const sinon = require( 'sinon' );
+const expect = chai.expect;
+const proxyquire = require( 'proxyquire' );
+
+describe( 'translations', () => {
+	describe( 'retryAsyncFunction()', () => {
+		let sandbox, stubs, retryAsyncFunction;
+		beforeEach( () => {
+			sandbox = sinon.sandbox.create();
+
+			mockery.enable( {
+				useCleanCache: true,
+				warnOnReplace: false,
+				warnOnUnregistered: false
+			} );
+
+			stubs = {
+				logger: {
+					info: sandbox.stub(),
+					warning: sandbox.stub(),
+					error: sandbox.stub()
+				}
+			};
+
+			retryAsyncFunction = proxyquire( '../../lib/translations/retryasyncfunction', {
+				'@ckeditor/ckeditor5-dev-utils': {
+					logger: () => stubs.logger
+				}
+			} );
+		} );
+
+		afterEach( () => {
+			mockery.disable();
+			sandbox.restore();
+		} );
+
+		it( 'should exeute an async function and return a promise', () => {
```

Comment:
You could check whether the value resolves by the promise (passed as a callback to `retryAsyncFunction`) is passed correctly.

```js
const result = retryAsyncFunction( () => Promise.resolve( 'Foo.' ) );

return result.then( value => expect( value ).to.equal( 'Foo.' ) );
```